storage/src/adb/any.rs

```diff
@@ -88,8 +80,17 @@ pub struct Any<E: RStorage + Clock + Metrics, K: Array, V: Array, H: CHasher> {
 
 impl<E: RStorage + Clock + Metrics, K: Array, V: Array, H: CHasher> Any<E, K, V, H> {
     /// Return an MMR initialized from `cfg`. Any uncommitted operations in the log will be
-    /// discarded and the state of the db will be as of the last committed operation.
-    pub async fn init(context: E, hasher: &mut H, cfg: Config) -> Result<Self, Error> {
+    /// discarded and the state of the db will be as of the last committed operation. If a bitmap is
+    /// provided, then a bit is appended for each operation in the operation log, with its value
+    /// reflecting its activity status. The bitmap is expected to already have a number of bits
+    /// corresponding to the portion of the database below the inactivity floor, and this method
+    /// will panic otherwise.
+    pub async fn init<const N: usize>(
```

Comment:
I thought we would introduce a new `adb` dialect for this rather than incorporating a `bitmap` here.

If you want to use some shared core for `adb::any` and `adb::latest` (maybe), I'd recommend still keep the `init` functions independent (i.e. I shouldn't see anything to do with a `BitMap` if I'm using `adb::any` unless we want to change its name to something like `adb::log` or `adb::mmr`)?

Reply:
Ultimately an ADB needs to know which operations in the update log are the active ones, and a bitmap is a natural data structure for indicating that. I think it makes sense to avoid Bitmap in the public API, but I think we'll need it or something like it (I could use the non-authenticated bitmap?) in an internal one to make this work without duplicating much of this code.

Reply:
Moved this out of the public API.